### PR TITLE
[TPU] Support multi-host inference

### PR DESCRIPTION
**Pull request by WoosukKwon**
Changing global rank and world size into local rank and local world size.

### CODE REVIEW

File: vllm/distributed/device_communicators/tpu_communicator.py [hunk 1]
```diff
@@ -18,9 +19,12 @@ def __init__(self, group: ProcessGroup):
             return
         self.disabled = False
 
-        local_rank = dist.get_rank(group)
-        world_size = dist.get_world_size(group)
-        pjrt.initialize_multiprocess(local_rank, world_size)
+        global_rank = dist.get_rank(group)
```
Comment (youkaichao) [inline vllm/distributed/device_communicators/tpu_communicator.py:22]: add a comment, saying that tpu tensor parallel size are determined at tpu pod slice creation time, and we only support using all of the tpus from a pod slice?

Comment (WoosukKwon) [reply vllm/distributed/device_communicators/tpu_communicator.py:25]: @youkaichao Good point. Added the comment. PTAL.

**Comment by youkaichao** [comment]
In ray gpu executor, there are these lines:

https://github.com/vllm-project/vllm/blob/7025b11d949b4efeb2584690c35f919c77027368/vllm/executor/ray_gpu_executor.py#L175-L191

to make sure the worker index aligns with machine boundary. you might need it in TPU, too. Otherwise local ranks can be wrong. for example, rank 0, 1, 2, 4 in one node, and 3, 5, 6, 7 in another node.

**Comment by WoosukKwon** [comment]
@youkaichao Can you explain more?

**Comment by youkaichao** [comment]
say you have 2 nodes, 8 TPUs in total.

ray actors are launched one by one. when you launch the first actor, it might live in node 0; when you launch the second actor, it might live in node 1. If you use the index of worker as global rank, then it will cause a problem.

**Comment by WoosukKwon** [comment]
@youkaichao Thanks for the explanation. Let me merge this PR first as there are users waiting for this and the scope of this PR is isolated to the TPU backend. I will address your comment in a followup PR.

**Pull request merged**